pinot-spi/src/main/java/org/apache/pinot/spi/config/table/StarTreeIndexConfig.java

```diff
@@ -34,6 +35,8 @@ public class StarTreeIndexConfig extends BaseJsonConfig {
   private final List<String> _skipStarNodeCreationForDimensions;
   // Function column pairs with delimiter "__", e.g. SUM__col1, MAX__col2, COUNT__*
   private final List<String> _functionColumnPairs;
+  // Compression mapping for function column pairs
+  private final Properties _functionColumnPairsConfig;
```

Comment:
Consider making it an explicit map
```suggestion
  private final Map<String, String> _compressionTypeMap;
```

You may also add a `_defaultCompressionType` so that we don't need to specify it for each individual functions

Reply:
I took the liberty of implementing a new `BaseJsonConfig` to handle this. It's backward compatible and potentially could substitute the `function__col` format making it more extendible.